lib/src/Base/Solver/RungeKutta.cxx

```diff
@@ -80,14 +81,24 @@ Sample RungeKutta::solve(const Point & initialState,
 }
 
 /* Perform one step of the RungeKutta method */
-Point RungeKutta::computeStep(const Scalar t,
+Point RungeKutta::computeStep(Function & transitionFunction,
+                              const Scalar t,
                               const Point & state,
                               const Scalar h) const
 {
-  const Point k1(transitionFunction_(t, state));
-  const Point k2(transitionFunction_(t + 0.5 * h, state + k1 * (0.5 * h)));
-  const Point k3(transitionFunction_(t + 0.5 * h, state + k2 * (0.5 * h)));
-  const Point k4(transitionFunction_(t + h, state + k3 * h));
+  Point parameter(1, t);
+  if (transitionFunction.getParameter().getDimension() > 0)
```

Comment:
This check looks strange. The parameter dimension should be equal to 1 and it should be checked only once in the constructor. Same remark for the Fehlberg class

Reply:
I thought there was an example with no parameter, but the tests seem to pass, strange.